### ISSUE TITLE
Add `Access-Control-Allow-Origin: *` to JSON file type headers?

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#7667]
Request from Makur:

> In order to remove grad.prod (D7 https://grad.prod.drupal.uiowa.edu/admin/content/file), we must be able to host JSON files for Workflow forms on D10 (https://grad.uiowa.edu/media/file/13246).
>
> In testing this, it looks like the new file is not being recognized by Workflow; it keeps spinning when the URL is changed from https://grad.prod.drupal.uiowa.edu/sites/gc/files/json/GradFellowProgramsV2_0.json to https://grad.uiowa.edu/sites/grad.uiowa.edu/files/2024-05/JSON.json. So, I’m wondering if the header for the D10 page meets the requirements. See highlighted below. Here are the requirements:
>
> JSON must be an array of "key"/"display" pairs. See https://workflow.uiowa.edu/choices/example for an example. 
The page that serves your JSON needs to have the header Access-Control-Allow-Origin: *. This allows cross domain access.
URL must be SSL (https://)
>
> If this is not possible, do you have any suggestions on where we can host the JSON files for Workflow forms?

**Comment by briand44** [comment]
D7 site has a few lines in .htaccess that sets the Header. Also seeing https://www.drupal.org/node/2715637.

**Comment by briand44** [comment]
Grooming conversation... Discussed trying to scope this to JSON files on the grad.uiowa.edu if possible. Not sure if the CORS method can scope this way and if that would even get invoked when calling a file directly.

**Comment by joewhitsitt** [comment]
Huh. I might have an intersect here with what I am trying to do for https://github.com/uiowa/uiowa/issues/7686.

>Access to XMLHttpRequest at 'https://emergency.stage.drupal.uiowa.edu/api/active' from origin 'https://signage.uiowa.edu.dd:8443' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by joewhitsitt** [comment]
Sean requested workflow editor access to be able to test https://github.com/uiowa/uiowa/pull/7698

**Comment by joewhitsitt** [comment]
I am trying to test #7698 against the json file extension but I don't think it is working. It works if *.json returns a 404 but not if it is the actual file. This makes me think that for files, we need to go the htaccess route:

```
<Files "*.json">
  Header set Access-Control-Allow-Origin "*"
</Files>
```

**Comment by joewhitsitt** [comment]
Found this old thread: https://iowaweb.slack.com/archives/C0164FNN1PV/p1675272202146099

```
<FilesMatch "\.json$">
  <IfModule mod_headers.c>
    Header set Access-Control-Allow-Origin "*"
  </IfModule>
</FilesMatch>
```
uigraduatecollege-dev/docroot/.htaccess

Ticket: https://acquia.my.site.com/s/case/5006g00000ji5tlAAA/appropriate-way-to-allow-json-file-access-cross-domain

Stackoverflow thread about security implications: https://stackoverflow.com/questions/12001269/what-security-risks-exist-when-setting-access-control-allow-origin-to-accept-all/56457665#56457665

**Comment by pyrello** [comment]
Fixed by #7698
